doc/whats_new/v1.5.rst

```diff
@@ -95,12 +95,16 @@ See :ref:`array_api` for more details.
 Support for building with Meson
 -------------------------------
 
-Meson is now supported as a build backend, see :ref:`Building from source
-<install_bleeding_edge>` for more details.
+From scikit-learn 1.5 onwards, Meson is the main supported way to build
+scikit-learn, see :ref:`Building from source <install_bleeding_edge>` for more
+details.
 
-:pr:`28040` by :user:`Loïc Estève <lesteve>`
+Unless we discover a major blocker, we intend to keep testing setuptools
+support until v1.6 and officially remove setuptools support when scikit-learn
+1.6 is released.
```

Comment:
maybe we can target 1.7 to be more inline with our usual 2 releases deprecation duration

Reply:
Does the type of build system that scikit-learn use effect users? I thought it was basically only people building from source or release managers who would notice the change? Which is why I think dropping in 1.6 would be Ok. Of course a bit more time for transitioning is also Ok.

Reply:
I guess that some people doing packaging themselves may be affected e.g. Linux distributions maybe?

Reply:
This is not a library API thingy so I think we can state that we will drop support for setuptools in 1.6.

If people complain, we can always extend this to 1.7 later.

Reply:
> maybe we can target 1.7 to be more inline with our usual 2 releases deprecation duration

I thought about that indeed, My current plan
1. rename `setup.py` to `_setup.py` in 1.6.dev0 (seemed to be the consensus during the meeting today)
2. in 1.6 you can rename `_setup.py` to `setup.py` to have one more release to actually move to Meson. Probably we still want to test setuptools during this period?
3. remove setuptools files completely in 1.7

I need to find again how Scipy did it but I think it was very reasonable, e.g. if you try to do `python setup.py something` it would tell you "setup.py is deprecated, please move to Meson and at last resort you still have the `_setup.py` option". What I am not sure about is whether this was guaranteed to work or more "this may work if you are lucky"

Reply:
I found the scipy way https://github.com/scipy/scipy/pull/19027/. They were still testing setuptools in 1.11.2 after `setup.py` -> `_setup.py` renaming and dropped it in 1.12.

Inspiration for the wording may also be used from the Scipy 1.9 release notes (a bit more detailed than what I currently have):
https://docs.scipy.org/doc/scipy/release/1.9.0-notes.html#scipy-switched-to-meson-as-its-build-system

Reply:
> This is not a library API thingy so I think we can state that we will drop support for setuptools in 1.6.

> If people complain, we can always extend this to 1.7 later.

Yeah thinking a bit more about this, this would be my preferred option too ...